app/controllers/ArrivalsController.scala

```diff
@@ -61,8 +59,12 @@ class ArrivalsController @Inject()(cc: ControllerComponents,
             case e: Upstream4xxResponse =>
```

Comment:
Could these case+if+else if be refactored into for comprehensions?

Reply:
There are plans to make 4xx and 5xx responses use status codes. They're currently exceptions. When this change occurs, it should improve the handling of downstream responses.